app/src/main/java/org/bspb/smartbirds/pro/service/DataOpsService.java

```diff
@@ -103,8 +107,26 @@ private void combineCommonWithEntries(Monitoring monitoring) {
                     //noinspection TryFinallyCanBeTryWithResources
                     try {
                         boolean firstLine = true;
+                        List<MonitoringEntry> entries = new ArrayList<>();
                         for (cursor.moveToFirst(); !cursor.isAfterLast(); cursor.moveToNext()) {
-                            MonitoringEntry entry = MonitoringManager.entryFromCursor(cursor);
+                            entries.add(MonitoringManager.entryFromCursor(cursor));
+                        }
+
+                        // Retain only keys available in all entries
+                        Set<String> normalizedKeys = new HashSet<>();
+                        for (MonitoringEntry entry : entries) {
+                            if (normalizedKeys.size() == 0) {
+                                // fill with initial values
+                                normalizedKeys.addAll(entry.data.keySet());
+                            }
+                            normalizedKeys.retainAll(entry.data.keySet());
```

Comment:
Will this remove empty optional fields as well?

Reply:
No, because the keys for the empty fields are present in the maps, so they won't be removed.